### PR TITLE
Describe show commands

### PR DESCRIPTION
**Pull request by YANG-DB**
### Description
Support PPL [Describe command](https://github.com/opensearch-project/sql/blob/main/docs/user/ppl/cmd/describe.rst)

### Issues Resolved
https://github.com/opensearch-project/opensearch-spark/issues/527

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLBasicITSuite.scala [hunk 1]
```diff
@@ -36,7 +38,42 @@ class FlintSparkPPLBasicITSuite
     }
   }
 
-  test("create ppl simple query test") {
+  test("describe table query test") {
+    val testTableQuoted = "`spark_catalog`.`default`.`flint_ppl_test`"
```
Comment (LantaoJin) [inline integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLBasicITSuite.scala:42]: This line should change to 
```
val testTableQuoted = "`flint_ppl_test`"
```

File: integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLBasicITSuite.scala [hunk 2]
```diff
@@ -36,7 +38,42 @@ class FlintSparkPPLBasicITSuite
...
+    Seq(testTable, testTableQuoted).foreach { table =>
+      val frame = sql(s"""
+           describe $table
```
Comment (LantaoJin) [inline integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLBasicITSuite.scala:45]: change to `describe flint_ppl_test`

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java [hunk 3]
```diff
@@ -107,6 +110,10 @@ public LogicalPlan visitExplain(Explain node, CatalystPlanContext context) {
 
     @Override
     public LogicalPlan visitRelation(Relation node, CatalystPlanContext context) {
+        if (node instanceof DescribeRelation) {
+            return context.with(new DescribeTableCommand(new TableIdentifier(node.getTableQualifiedName().toString()), null, false, seq()));
```
Comment (LantaoJin) [inline ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java:114]: And here:
```
            return context.with(
                new DescribeTableCommand(
                    new TableIdentifier(node.getTableQualifiedName().toString()),
                    scala.collection.immutable.Map$.MODULE$.<String, String>empty(),
                    false,
                    DescribeRelation$.MODULE$.getOutputAttrs()));
```

Comment (YANG-DB) [reply ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java]: thanks!!

File: integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLBasicITSuite.scala [hunk 4]
```diff
@@ -36,7 +38,42 @@ class FlintSparkPPLBasicITSuite
...
+        Row("month", "int", null)
+      )
+      // Compare the results
+      implicit val rowOrdering: Ordering[Row] = Ordering.by[Row, String](_.getAs[String](0))
+      assert(results.sorted.sameElements(expectedResults.sorted))
+      // Retrieve the logical plan
+      val logicalPlan: LogicalPlan = frame.queryExecution.commandExecuted.asInstanceOf[CommandResult].commandLogicalPlan
+      // Define the expected logical plan
+      val expectedPlan: LogicalPlan =
+          DescribeTableCommand(TableIdentifier("default.flint_ppl_test"), null, isExtended = false, output = Seq())
```
Comment (LantaoJin) [inline integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLBasicITSuite.scala:70]: @YANG-DB to make this IT test successful.
The expectedPlan should change as following either.
```
      val expectedPlan: LogicalPlan =
        DescribeTableCommand(
          TableIdentifier("flint_ppl_test"),
          Map.empty[String, String],
          isExtended = false,
          output = DescribeRelation.getOutputAttrs)
```

File: integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLBasicITSuite.scala [hunk 5]
```diff
@@ -36,7 +38,46 @@ class FlintSparkPPLBasicITSuite
     }
   }
 
-  test("create ppl simple query test") {
+  test("describe table query test") {
+    val testTableQuoted = "`spark_catalog`.`default`.`flint_ppl_test`"
+    Seq(testTable, testTableQuoted).foreach { table =>
+      val frame = sql(s"""
+           describe flint_ppl_test
```
Comment (LantaoJin) [inline integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLBasicITSuite.scala:45]: Oh, we miss testing the table_identifier with quoted.
```
Seq("flint_ppl_test", "`flint_ppl_test`").foreach { table =>
  val frame = sql(s"""
     describe $table
```

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java [hunk 6]
```diff
@@ -107,6 +111,15 @@ public LogicalPlan visitExplain(Explain node, CatalystPlanContext context) {
 
     @Override
     public LogicalPlan visitRelation(Relation node, CatalystPlanContext context) {
+        if (node instanceof DescribeRelation) {
+            return context.with(
+                    new DescribeTableCommand(
+                            new TableIdentifier(node.getTableQualifiedName().toString()),
```
Comment (LantaoJin) [inline ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java:117]: I checked the spark sql reference. The DESCRIBE command syntax is `DESCRIBE [ database_name. ] table_name`.
So to support database prefix, here need additional modification:
```
            TableIdentifier identifier;
            if (node.getTableQualifiedName().getParts().size() == 1) {
                identifier = new TableIdentifier(node.getTableQualifiedName().getParts().get(0));
            } else {
                identifier = new TableIdentifier(
                    node.getTableQualifiedName().getParts().get(1),
                    Option$.MODULE$.apply(node.getTableQualifiedName().getParts().get(0)));
            }
            return context.with(
                new DescribeTableCommand(
                    identifier,
                    scala.collection.immutable.Map$.MODULE$.<String, String>empty(),
                    false,
                    DescribeRelation$.MODULE$.getOutputAttrs()));
```

File: integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLBasicITSuite.scala (hunk 5 above)
Comment (YANG-DB) [reply integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLBasicITSuite.scala:45]: yes probably need to add some more fail based use cases...

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java (hunk 6 above)
Comment (LantaoJin) [reply ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java:117]: Then we can test on Seq("flint_ppl_test", "`flint_ppl_test`", "default.flint_ppl_test", "`default`.`flint_ppl_test`")

Comment (YANG-DB) [reply ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java:117]: @LantaoJin thanks again

**Pull request closed without merging**